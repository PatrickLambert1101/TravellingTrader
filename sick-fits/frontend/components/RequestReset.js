import React, { useState } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { useFormState } from 'react-use-form-state';
import Error from './ErrorMessage';
import Form from './styles/Form';
const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;

export default function() {
  const [formState, { email }] = useFormState();

  // const saveToState = e => {
  //   this.setState({ [e.target.name]: e.target.value });
  // };

  return (
    <Mutation mutation={REQUEST_RESET_MUTATION} variables={email}>
      {(reset, { error, loading, called }) => {
        return (
          <Form
            data-test="form"
            method="post"
            onSubmit={async e => {
              e.preventDefault();
              await reset();
              email('');
            }}
          >
            <fieldset disabled={loading} aria-busy={loading}>
              <h2>Reset your password</h2>
              <Error error={error} />
              {!error && !loading && called && (
                <p>Success, check your email bro</p>
              )}
              <label htmlFor="email">
                Email
                <input {...email('email')} />
              </label>

              <button type="submit">Request a reset!</button>
            </fieldset>
          </Form>
        );
      }}
    </Mutation>
  );
}

export { REQUEST_RESET_MUTATION };
