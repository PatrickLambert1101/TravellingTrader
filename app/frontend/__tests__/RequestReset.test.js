import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, {
  REQUEST_RESET_MUTATION
} from '../components/RequestReset';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'lambertpatrick09@gmail.com' }
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } }
    }
  }
];

describe('<RequestReset /> ', () => {
  it('renders and matches snapshot', () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    expect(toJSON(wrapper.find('form[data-test="form"]'))).toMatchSnapshot();
  });

  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    //simulate typing an email
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'lambertpatrick09@gmail.com' }
    });
    wrapper.find('form').simulate('submit');
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find('form[data-test="form"]'))).toMatchSnapshot();
  });
});
