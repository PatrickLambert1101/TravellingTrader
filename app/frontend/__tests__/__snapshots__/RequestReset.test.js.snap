// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset />  calls the mutation 1`] = `
<form
  className="Form-sc-1h5xw8s-0 jIreny"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={true}
    disabled={true}
  >
    <h2>
      Reset your password
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value="lambertpatrick09@gmail.com"
      />
    </label>
    <button
      type="submit"
    >
      Request a reset!
    </button>
  </fieldset>
</form>
`;

exports[`<RequestReset />  renders and matches snapshot 1`] = `
<form
  className="Form-sc-1h5xw8s-0 jIreny"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Reset your password
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request a reset!
    </button>
  </fieldset>
</form>
`;
